### PR TITLE
http: making the admission control filter a dual filter

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: integration tested
Docs Changes: n/a
Release Notes: inline
Part of https://github.com/envoyproxy/envoy/issues/10455

### CODE REVIEW

**Comment by alyssawilk** [comment]
Hey @phlax can you look at this vs prior push: one failed format due to flags errors and the other succeeded
CodeChecker ERROR [runtime_guards] Missing from changelogs: envoy_reloadable_features_allow_upstream_filters
reordering shouldn't break or fix so I think there's a bug in the checks

File: source/extensions/filters/http/common/factory_base.h [hunk 1]
```diff
@@ -87,10 +87,11 @@ class DualFactoryBase : public CommonFactoryBase<ConfigProto, RouteConfigProto>,
 
   struct DualInfo {
     DualInfo(Server::Configuration::UpstreamHttpFactoryContext& context)
-        : init_manager(context.initManager()) {}
+        : init_manager(context.initManager()), scope(context.scope()) {}
     DualInfo(Server::Configuration::FactoryContext& context)
-        : init_manager(context.initManager()) {}
+        : init_manager(context.initManager()), scope(context.scope()) {}
     Init::Manager& init_manager;
+    Stats::Scope& scope;
```
Comment (wbpcode) [inline source/extensions/filters/http/common/factory_base.h:94]: would split this to a separated minor PR be better?

**Comment by wbpcode** [comment]
Awesome. I had no idea that upstream filters would move so fast. 👍 Thank you so much.

**Review by mattklein123 — APPROVED** [review]
Nice!

**Pull request merged**